src/utils/connect.js

```diff
@@ -0,0 +1,115 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+import * as React from 'react';
+import { connect } from 'react-redux';
+import type {} from 'react-redux';
+import type { Dispatch, State } from '../types/store';
+
+type MapStateToProps<OwnProps: Object, StateProps: Object> = (
+  state: State,
+  ownProps: OwnProps
+) => StateProps;
+
+type MapDispatchToProps<OwnProps: Object, DispatchProps: Object> =
+  | ((dispatch: Dispatch, ownProps: OwnProps) => DispatchProps)
+  | DispatchProps;
+
+type MergeProps<
+  StateProps,
+  DispatchProps: Object,
+  OwnProps: Object,
+  Props: Object
+> = (
+  stateProps: StateProps,
+  dispatchProps: DispatchProps,
+  ownProps: OwnProps
+) => Props;
+
+type ConnectOptions = {
+  pure?: boolean,
+  areStatesEqual?: boolean,
+  areOwnPropsEqual?: boolean,
+  areStatePropsEqual?: boolean,
+  areMergedPropsEqual?: boolean,
+  storeKey?: boolean,
+  withRef?: boolean,
+};
+
+export type ExplicitConnectOptions<
+  OwnProps: Object,
+  StateProps: Object,
+  DispatchProps: Object
```

Comment:
given how often we don't use `OwnProps` I would have put it at the end of the parameter list, with a default value `{||}`, so that we don't have to specify it when we don't need it.

Now is it worth changing everything you did? I think yes as this simplifies some things, but your call.

Reply:
Good call. I can hit this with one other change I want to do.

Reply:
Actually looking at this a second time, I think I would prefer them in the order they are in, as it's the order they are applied. e.g. `StateProps` override `OwnProps`.